### PR TITLE
Rework Text Field and Text Area states

### PR DESCRIPTION
**Pull request by vladmoroz**
See changelog.

### CODE REVIEW

File: packages/radix-ui-themes/changelog.md [hunk 1]
```diff
@@ -7,6 +7,18 @@
   - Remove comments from the CSS build.
   - Make sure that forced light/dark appearance on the `Theme` component also sets the corresponding browser colors, like the correct input autofill background color.
   - Use `outline` rather than `box-shadow` for most focus styles, which avoids a slight anti-aliasing issue in Chrome on focused elements
+- `TextArea`, `TextField`
+  - Refine padding values for a more balanced look.
+    - Use `text-indent` instead of `padding-left` so that long values aren't truncated on the left when the cursor is at the end of the input.
```
Comment (vladmoroz) [inline packages/radix-ui-themes/changelog.md:12]: Left edge not clipped:

Screenshot 2023-09-04 at 12 35 08

File: packages/radix-ui-themes/changelog.md [hunk 2]
```diff
@@ -7,6 +7,18 @@
...
+  - Refine how the inner shadows are applied so that they blend with the container background.
```
Comment (vladmoroz) [inline packages/radix-ui-themes/changelog.md:13]: https://twitter.com/vladyslavmoroz/status/1698282797830484330

File: packages/radix-ui-themes/changelog.md [hunk 3]
```diff
@@ -7,6 +7,18 @@
...
+  - Refine and normalise the look and feel of disabled and read-only states.
+    - Improve the background colors in dark mode.
+    - Improve the appearance of placeholders.
+    - Remove visual distinctions between disabled and read-only states.
+    - Keep the differences between the variants when the inputs are disabled or read-only.
+    - Fix a Safari bug when the text value would appear tinted in the disabled input.
+  - Remove ellipsis truncation, as this prevented long values from being shown when scrolling on the input horizontally in Chrome.
+  - Improve autofill styles.
```
Comment (vladmoroz) [inline packages/radix-ui-themes/changelog.md:21]: Before:

Screenshot 2023-09-04 at 10 39 27

After:

Screenshot 2023-09-04 at 10 39 37

File: packages/radix-ui-themes/changelog.md [hunk 4]
```diff
@@ -7,6 +7,18 @@
...
+- `TextArea`, `TextField`
+  - Refine padding values for a more balanced look.
+    - Use `text-indent` instead of `padding-left` so that long values aren't truncated on the left when the cursor is at the end of the input.
+  - Refine how the inner shadows are applied so that they blend with the container background.
+  - Refine and normalise the look and feel of disabled and read-only states.
+    - Improve the background colors in dark mode.
+    - Improve the appearance of placeholders.
+    - Remove visual distinctions between disabled and read-only states.
+    - Keep the differences between the variants when the inputs are disabled or read-only.
+    - Fix a Safari bug when the text value would appear tinted in the disabled input.
```
Comment (vladmoroz) [inline packages/radix-ui-themes/changelog.md:19]: 🫠

Screenshot 2023-09-04 at 11 37 52

File: packages/radix-ui-themes/changelog.md [hunk 5]
```diff
@@ -7,6 +7,18 @@
...
   - Make sure that forced light/dark appearance on the `Theme` component also sets the corresponding browser colors, like the correct input autofill background color.
   - Use `outline` rather than `box-shadow` for most focus styles, which avoids a slight anti-aliasing issue in Chrome on focused elements
+- `TextArea`, `TextField`
+  - Refine padding values for a more balanced look.
+    - Use `text-indent` instead of `padding-left` in the `TextField` so that long values aren't truncated on the left when the cursor is at the end of the input.
+  - Refine how the inner shadows are applied so that they blend with the container background.
+  - Refine and normalise the look and feel of disabled and read-only states.
+    - Improve the background colors in dark mode.
+    - Improve the appearance of placeholders.
+    - Remove visual distinctions between disabled and read-only states.
```
Comment (vladmoroz) [inline packages/radix-ui-themes/changelog.md:17]: The users don't care imo and the differences were superficial and hard to reason about much

**Comment by needim** [comment]
Amazing PR! Thanks. 🚀

**Pull request merged**